### PR TITLE
Country code lookup by synonym

### PR DESCRIPTION
**Pull request by davidmthornton**
Logic to try to find country code from synonyms resource if an invalid FCO country name received from GRS.

Tidied a few warnings as well.

### CODE REVIEW

File: conf/resources/countrySynonyms.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "GB": {
+    "synonyms": [
+        "England", "Scotland", "Wales"
```
Comment (PSouthern) [inline conf/resources/countrySynonyms.json:4]: Should there be a Northern Ireland synonym, or has this intentionally been omitted?

**Review by ajchilts — APPROVED** [review]
Once Pauls's question has been answered. 👍

File: conf/resources/countrySynonyms.json (hunk 1 above)
Comment (davidmthornton) [reply conf/resources/countrySynonyms.json:4]: GB doesn't technically include Northern Ireland, but in countriesEN its official name is listed as "The United Kingdom of Great Britain and Northern Ireland". Looking [here](https://find-and-update.company-information.service.gov.uk/advanced-search/get-results?companyNameIncludes=&companyNameExcludes=&registeredOfficeAddress=Northern+Ireland&incorporationFromDay=&incorporationFromMonth=&incorporationFromYear=&incorporationToDay=&incorporationToMonth=&incorporationToYear=&sicCodes=&dissolvedFromDay=&dissolvedFromMonth=&dissolvedFromYear=&dissolvedToDay=&dissolvedToMonth=&dissolvedToYear=) there is a menagerie of different combinations.

Comment (davidmthornton) [reply conf/resources/countrySynonyms.json:4]: I'd be up for adding "Northern Ireland" to the GB synonyms though as it's bound to happen.

**Review by liampan — APPROVED** [review]

**Pull request merged**